lib/Installation/SystemRole/Sle/SystemRolePage.pm

```diff
@@ -0,0 +1,34 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The module provides interface to act on System Role page in
+#          the installer.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::SystemRole::Sle::SystemRolePage;
+use parent Installation::SystemRole::SystemRolePage;
+use strict;
+use warnings;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init();
+}
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
```

Comment:
let's remove this super call to init here (we can still use this class as a child to not repeat methods), because the roles from openSUSE do not apply (check the other comment below) so we can list here the ones for SLE in a more clear way.

Reply:
Not sure if you had something else in mind, but if I remove this call, methods are not properly passed from parent : http://waaa-amazing.suse.cz/tests/15827#step/validate_selected_role_text_mode/2

Reply:
ahm I see, now you changed the next functionality, we don't initialize the button without this line above. The problem that I see is that we will accumulate both roles in the same object, when we should completely override them.
The methods are properly passed, this call only call the parent method for init, the rest of the methods `get_selected_roles`, etc. are passed.
The more elegant solution would be to create abstract class to not repeat the method, so that abstract class inherit from NavigationBase, and then both SystemRolePage just override the constructor.
Another way would be to do something like this `$self->{role}{GNOME_desktop} = 'SLES with GNOME';` so we override the property {role} but this is a bit more obscure.
Anyway I'm fine with this solution, we are supposed to know what to select from test layer, so this is more like technical implementation detail that I want to share with you, perhaps for next time.

Reply:
Thanks for the ideas :)